src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -947,16 +945,40 @@ class PartitionClauseSegment(BaseSegment):
 
 @ansi_dialect.segment()
 class FrameClauseSegment(BaseSegment):
-    """A frame clause for window functions."""
+    """A frame clause for window functions.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_Window_function_synopsis.html
+    """
 
     type = "frame_clause"
-    match_grammar = StartsWith("ROWS")
-    # TODO: Expand a parse statement here properly to actually
-    # parse rather than assuming that it's good.
-    # parse_grammar = Sequence(
-    #    'ROWS',
-    #    ...
-    # )
+    match_grammar = Sequence(
+        "ROWS",
```

Comment:
Should we support `RANGE` and `GROUPS` too?

https://www.postgresql.org/docs/13/sql-expressions.html

Reply:
Ahh okay, this is bigger than I thought. It varies by dialect. I'll rewrite for ANSI and Postgres

Reply:
It's much beefier now